app/src/main/java/org/schabi/newpipe/fragments/list/kisok/KioskFragment.java

```diff
@@ -0,0 +1,148 @@
+
+    private View headerRootLayout;
+    private TextView headerTitleView;
+
+    public static KioskFragment getInstance(int serviceId)
+            throws ExtractionException {
+        KioskFragment instance = new KioskFragment();
+        StreamingService service = NewPipe.getService(serviceId);
+        String defaultKioskType = service.getKioskList().getDefaultKioskType();
+        UrlIdHandler defaultKioskTypeUrlIdHandler = service.getKioskList()
+                .getUrlIdHandlerByType(defaultKioskType);
+        instance.setInitialData(serviceId,
+                defaultKioskTypeUrlIdHandler.getUrl(defaultKioskType),
+                defaultKioskType);
+        return instance;
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // LifeCycle
+    //////////////////////////////////////////////////////////////////////////*/
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        return inflater.inflate(R.layout.fragment_kiosk, container, false);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Init
+    //////////////////////////////////////////////////////////////////////////*/
+
+    protected View getListHeader() {
+        headerRootLayout = activity.getLayoutInflater().inflate(R.layout.kiosk_header, itemsList, false);
+        headerTitleView = headerRootLayout.findViewById(R.id.kiosk_title_view);
+
+        return headerRootLayout;
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Load and handle
+    //////////////////////////////////////////////////////////////////////////*/
+
+    @Override
+    public Single<KioskInfo> loadResult(boolean forceReload) {
+        String contentCountry = PreferenceManager
+                .getDefaultSharedPreferences(activity)
+                .getString(getString(R.string.search_language_key),
+                        getString(R.string.default_language_value));
+        return ExtractorHelper.getKioskInfo(serviceId, url, contentCountry, forceReload);
+    }
+
+    @Override
+    public Single<ListExtractor.NextItemsResult> loadMoreItemsLogic() {
+        return ExtractorHelper.getMoreKisokItems(serviceId, url, currentNextItemsUrl);
+    }
+
+    /*//////////////////////////////////////////////////////////////////////////
+    // Contract
+    //////////////////////////////////////////////////////////////////////////*/
+
+    @Override
+    public void showLoading() {
+        super.showLoading();
+        animateView(headerRootLayout, false, 200);
+        animateView(itemsList, false, 100);
+    }
+
+    @Override
+    public void handleResult(@NonNull final KioskInfo result) {
+        super.handleResult(result);
+
+        animateView(headerRootLayout, true, 100);
+        headerTitleView.setText(result.type);
```

Comment:
You are right. Id/type/name was not meant to be displayed directly.
I'll add a todo.

Reply:
Yeah, I was thinking about it, but I couldn't find a better way than bunch of `if` statements, so do you have any better ideas?

Reply:
I see. Isn't it possible to get some stringresource like R.string.<some_string> programmatically? Something like `Resource.getString("some_string");` If not than we might have to find another solution.

Reply:
I thought in doing a "generify" and use the most common term as the id/type, because it is more convenient to use a generic term to cover a lot of cases. 

_Example:_ SoundCloud uses "New & Hot", YouTube's equivalent is "Trending".
In this case, "Trending" is the selected one (as it is more known).

_Example 2:_ A lot of services use "Top ##", this can easily be "generified" because only the number changes.

###### Though there's a downside to it, we lost the "uniqueness" of each service.

Reply:
I don't know I don't like generalizing either.
But why don't we just crawl the Name.

Reply:
I thought in generify the extractor and then _**in the NewPipe itself**_, make a unique translation for each service, much better than hardcode-checking in this repository (and translating in the extractor repository would force us to use some way to get the translations).

E.g.:
```java
switch (kiosk.getType()) {
    case Top:
        getString(soundcloud_top); // Top
    case Trending:
        getString(soundcloud_trending); // New & Hot
}
```

>But why don't we just crawl the Name

From where? SoundCloud for example, doesn't have all languages that NewPipe support.

Reply:
Yeah, I don't like 'generifying' it. The problem with SoundCloud is also that they also have separate Top 50's and New & Hot's for different genres, which we don't have implemented yet though.

Reply:
Well I don't like it ether. Because many services might make thins different form other services, even tho they might almost do "similar" things. Also I'm 100% sure people who will impelment their own NewPipe fronted will get that wrong. In my opinion It's the isiest solution to just crawl the translation. If it's not available in the language you selected then the content of the service will bin in the fall-back language anyway, so taking the effort ant translating only the headline would not be worth the work.

Reply:
I've add [KioskTranslator](https://github.com/TeamNewPipe/NewPipe/blob/f7534b3a0f72ca86f195de2a17e1bbab4669a74b/app/src/main/java/org/schabi/newpipe/util/KioskTranslator.java) with which local translation is possible.